front/src/main.js

```diff
@@ -0,0 +1,423 @@
+// The Vue build version to load with the `import` command
+// (runtime-only or standalone) has been set in webpack.base.conf with an alias.
+import Vue from 'vue'
+import Vuex from 'vuex'
+import VueRouter from 'vue-router'
+import VueTimeago from 'vue-timeago'
+
+import App from './App'
+import GameList from './components/GameList'
+import TeamList from './components/TeamList'
+import MatchList from './components/MatchList'
+import TournamentList from './components/TournamentList'
+import Tournament from './components/Tournament'
+import Match from './components/Match'
+import Login from './components/Login'
+import Dashboard from './components/Dashboard'
+import Team from './components/Team'
+import auth from './auth'
+import api from './api'
+
+Vue.use(Vuex)
+Vue.use(VueRouter)
+
+Vue.use(VueTimeago, {
+  name: 'timeago',
+  locale: 'en-US',
+  locales: {
+    'en-US': require('vue-timeago/locales/en-US.json')
+  },
+})
+
+const routes = [
+  {
+    path: '/',
+    redirect: '/dashboard',
+  },
+  {
+    path: '/dashboard',
+    component: Dashboard,
```

Comment:
You can resolve those components asynchronously. This will enable webpack to create chunks based on dependencies. http://router.vuejs.org/en/advanced/lazy-loading.html